tutorial/creating_packages/test_conan_packages.rst

```diff
@@ -1,2 +1,171 @@
+
+
+In all the previous sections of the tutorial, we used the *test_package*. It was invoked
+automatically at the end of the ``conan create`` command after building our package
+verifying that the package is created correctly. Let’s explain the *test_package* in more
+detail in this section:
+
+Please, first clone the sources to recreate this project. You can find them in the
+`examples2.0 repository <https://github.com/conan-io/examples2>`_ on GitHub:
+
+.. code-block:: bash
+
+    $ git clone https://github.com/conan-io/examples2.git
+    $ cd examples2/tutorial/creating_packages/testing_packages
+
+
+Some important notes to have in mind about the *test_package*:
+
+* The *test_package* folder is different from unit or integration tests. These tests are
+  “package” tests, and validate that the package is properly created, and that the package
+  consumers will be able to link against it and reuse it.
+
+* It is a small Conan project itself, it contains its own *conanfile.py*, and its source
+  code including build scripts, that depends on the package being created, and builds and
+  execute a small application that requires the library in the package.
+
+* It doesn’t belong to the package. It only exist in the source repository, not in the
+  package.
+
+The *test_package* folder for for our hello/1.0 Conan package has the following contents:
+
+.. code-block:: text
+
+   test_package
+    ├── CMakeLists.txt
+    ├── conanfile.py
+    └── src
+        └── example.cpp
+
+Let's have a look at the different files that are part of the *test_package*. First, the
+*example.cpp* is just a minimal example of how to use the *libhello* library that we are
+packaging:
+
+.. code-block:: cpp
+    :caption: *test_package/src/example.cpp*
+
+    #include "hello.h"
+
+    int main() {
+        hello();
+    }
+
+Then the *CMakeLists.txt* to tell CMake how to build the example:
+
+.. code-block:: cpp
+    :caption: *test_package/src/example.cpp*
+
+    cmake_minimum_required(VERSION 3.15)
+    project(PackageTest CXX)
+
+    find_package(hello CONFIG REQUIRED)
+
+    add_executable(example src/example.cpp)
+    target_link_libraries(example hello::hello)
+
+Finally, the recipe for the *test_package* that consumes the *hello/1.0* Conan package:
+
+.. code-block:: python
+    :caption: *test_package/conanfile.py*
+
+    import os
+
+    from conan import ConanFile
+    from conan.tools.cmake import CMake, cmake_layout
+    from conan.tools.build import cross_building
+
+
+    class helloTestConan(ConanFile):
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "CMakeDeps", "CMakeToolchain"
+
+        def requirements(self):
+            self.requires(self.tested_reference_str)
+
+        def build(self):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+
+        def layout(self):
+            cmake_layout(self)
+
+        def test(self):
+            if not cross_building(self):
```

Comment:
Maybe use better ``can_run()`` here? There are things that are cross-build, but still can run, like Mac with different architectures using emulation?